io/disk/lvsetup.py

```diff
@@ -102,15 +109,26 @@ def create_lv(self):
         A volume group with given name is created in the ramdisk. It then
         creates a logical volume.
         """
-        self.ramdisks.append(lv_utils.vg_ramdisk(self.disk, self.vg_name,
-                                                 self.ramdisk_vg_size,
-                                                 self.ramdisk_basedir,
-                                                 self.ramdisk_sparse_filename))
         lv_utils.lv_create(self.vg_name, self.lv_name, self.lv_size)
         lv_utils.lv_mount(self.vg_name, self.lv_name, self.mount_loc,
                           create_filesystem=self.fs_name)
         lv_utils.lv_umount(self.vg_name, self.lv_name)
 
+    @avocado.fail_on(lv_utils.LVException)
+    def delete_lv(self):
+        """
+        Clear all PV,VG, LV and snapshots created by the test.
+        """
+        # Remove created VG and unmount from base directory
+        errs = []
+        for ramdisk in self.ramdisks:
+            try:
+                lv_utils.vg_ramdisk_cleanup(*ramdisk)
+            except Exception as exc:
+                errs.append("Fail to cleanup ramdisk %s: %s" % (ramdisk, exc))
+        if errs:
+            self.fail("\n".join(errs))
```

Comment:
I feel exception if enough here, and not require the error again.
looks like doing the same thing twice.

Reply:
@Naresh-ibm We are just catching the exception and failing, not throwing.
Also, that is the existing code which I am just creating a function for, not new code.